runners/google-cloud-dataflow-java/worker/build.gradle

```diff
@@ -85,24 +87,34 @@ dependencies {
   compile google_api_services_dataflow
   compile library.java.avro
   compile library.java.google_api_client
+  permitUnusedDeclared library.java.google_api_client // BEAM-11761
+  compile library.java.google_auth_library_credentials
   compile library.java.google_http_client
   compile library.java.google_http_client_jackson2
   compile library.java.jackson_annotations
   compile library.java.jackson_core
   compile library.java.jackson_databind
   compile library.java.joda_time
+  compile library.java.proto_google_common_protos
   shadow library.java.vendored_guava_26_0_jre
   compile library.java.slf4j_api
+  compile "io.opencensus:opencensus-api:0.28.0"
+  compile "org.conscrypt:conscrypt-openjdk-uber:2.5.1"
+  compile "com.google.guava:guava:25.1-jre"
   compile "javax.servlet:javax.servlet-api:3.1.0"
 
   // Conscrypt shouldn't be included here because Conscrypt won't work when being shaded.
   // (Context: https://github.com/apache/beam/pull/13846)
   // Conscrypt will be added to runtime dependencies by GrpcVendoring so compileOnly works for now.
   compileOnly "org.conscrypt:conscrypt-openjdk:2.5.1:linux-x86_64"
+  permitUnusedDeclared "org.conscrypt:conscrypt-openjdk:2.5.1:linux-x86_64" // BEAM-11761
```

Comment:
It looks like this dependency was modified recently. @veblush @kennknowles I couldn't follow the reasoning behind #13846. I have two questions here:

1. How are `conscrypt-openjdk` and `conscrypt-openjdk-uber` related?
2. Why do we need `conscrypt-openjdk`, since it's compileOnly yet the plugin is telling us we don't need it for Java compilation?

Reply:
1. [conscrypt-openjdk-uber](https://github.com/google/conscrypt#uber-jar) is a collection of jars for all supported platform so `conscrypt-openjdk` is a subset of `conscrypt-openjdk-uber`. Presumably `conscrypt-openjdk` was chosen to reduce the size of merged jar because it's required to support linux only. If we doesn't need it to be merged, using `conscrypt-openjdk-uber` might be acceptable as well.
2. AFAIK, following is the one depends on conscrypt in worker jar. https://github.com/apache/beam/blob/644cf47bc7d48256014164077b1d13401a5d8c3a/runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowWorkerHarnessHelper.java#L72

Reply:
Thanks for the info @veblush.

My understanding is that `conscrypt-openjdk-uber` is transitively being added to the worker's compile classpath through the dependency on `:model:pipeline` etc., which seems to get it through `GrpcVendoring_1_26_0`. Because of this, the plugin thinks `org.conscrypt.OpenSSLProvider` is being imported from `conscrypt-openjdk-uber`, not `conscrypt-openjdk`.

So I think we should add `conscrypt-openjdk-uber` as a `compileOnly` dependency, and remove `conscrypt-openjdk`. WDYT?

Reply:
I think it would work.

Reply:
> My plan is to keep the compileOnly line added by PR #13846 but change openjdk to openjdk-uber in that line and leave the comment block alone.

Sounds good.

> @veblush I would guess the linux-x86_64 version was chosen because all jars under https://repo.maven.apache.org/maven2/org/conscrypt/conscrypt-openjdk/2.5.1/ are OS specific and we had to just pick one?

Dataflow workers only run on Linux VMs, so it only needs Linux. But vendored GRPC needs to support all platforms, so it includes the "uber" jar.